sharepoint-adfs/azuredeploy.json

```diff
@@ -0,0 +1,855 @@
+      }
+    },
+    "vmSPName": {
+      "type": "string",
+      "defaultValue": "SP",
+      "minLength": 2,
+      "metadata": {
+        "description": "Internal name of SP VM"
+      }
+    },
+    "vmSPSize": {
+      "type": "string",
+      "defaultValue": "Standard_A8_v2",
+      "metadata": {
+        "description": "Size of the SQL VM"
+      }
+    },
+    "vmSPPublicIPDnsName": {
+      "type": "string",
+      "minLength": 2,
+      "metadata": {
+        "description": "'Input' used to create public DNS name 'Input.region.cloudapp.azure.com' of SP VM"
+      }
+    },
+    "vmsTimeZone": {
+      "type": "string",
+      "defaultValue": "Central European Standard Time",
+      "minLength": 2,
+      "metadata": {
+        "description": "Time zone of the VMs. Get the list with '[System.TimeZoneInfo]::GetSystemTimeZones().Id'. Note that 'UTC' works but 'UTC+xx' does NOT work."
+      }
+    },
+    "dscDCTemplateURL": {
+      "type": "string",
+      "defaultValue": "https://raw.githubusercontent.com/Azure/azure-quickstart-templates/master/sharepoint-adfs/dsc/ConfigureDCVM.zip",
+      "metadata": {
+        "description": "Link to DSC configuration of DC VM"
+      }
+    },
+    "dscSQLTemplateURL": {
+      "type": "string",
+      "defaultValue": "https://raw.githubusercontent.com/Azure/azure-quickstart-templates/master/sharepoint-adfs/dsc/ConfigureSQLVM.zip",
+      "metadata": {
+        "description": "Link to DSC configuration of SQL VM"
+      }
+    },
+    "dscSPTemplateURL": {
+      "type": "string",
+      "defaultValue": "https://raw.githubusercontent.com/Azure/azure-quickstart-templates/master/sharepoint-adfs/dsc/ConfigureSPVM.zip",
+      "metadata": {
+        "description": "Link to DSC configuration of SP VM"
+      }
+    }
+  },
+  "variables": {
+    "storageAccountName": "[concat(parameters('templatePrefix'), 'st')]",
+    "vNetPrivateName": "[concat(parameters('templatePrefix'), '-vnet')]",
+    "generalSettings": {
+      "vmsStorageAccountContainerName": "vhds",
+      "vmsVnetID": "[resourceId('Microsoft.Network/virtualNetworks', variables('vNetPrivateName'))]",
+      "vNetPrivatePrefix": "10.0.0.0/16",
+      "vNetPrivateSubnet1Name": "Subnet-1",
+      "vNetPrivateSubnet1Prefix": "10.0.1.0/24",
+      "vNetPrivateSubnet2Name": "Subnet-2",
+      "vNetPrivateSubnet2Prefix": "10.0.2.0/24",
+      "vNetPrivateSubnet3Name": "Subnet-3",
+      "vNetPrivateSubnet3Prefix": "10.0.3.0/24"
+    },
+    "vmDC": {
+      "vmImagePublisher": "MicrosoftWindowsServer",
+      "vmImageOffer": "WindowsServer",
+      "vmImageSKU": "2016-Datacenter",
+      "vmOSDiskName": "vm-DC-OSDisk",
+      "vmDataDiskName": "vm-DC-DataDisk",
+      "vmVmSize": "[parameters('vmDCSize')]",
+      "vmSubnetRef": "[concat(variables('generalSettings').vmsVnetID, '/subnets/', variables('generalSettings').vNetPrivateSubnet1Name)]",
```

Comment:
vmSubnetRef uses variables('generalSettings').vmsVnetID, which uses resourceId, is it what you meant?